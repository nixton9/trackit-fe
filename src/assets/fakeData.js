export const notes = [
  {
    id: 1,
    title: 'Lista de compras',
    date: '2020-07-20',
    content: 'asdasdad',
    tags: [
      { id: 1, name: 'Primary', color: '#F44336' },
      { id: 2, name: 'Secondary', color: '#E91E63' }
    ]
  },
  {
    id: 2,
    title: 'Apontamentos aula',
    date: '2020-06-25',
    content: 'asdasdad',
    tags: [
      { id: 3, name: 'Studies', color: '#607D8B' },
      { id: 4, name: 'School', color: '#00BCD4' }
    ]
  },
  {
    id: 3,
    title: 'Random shit',
    date: '2020-07-15',
    content: 'asdasdad',
    tags: [{ id: 5, name: 'Random', color: '#3F51B5' }]
  },
  {
    id: 4,
    title: 'Work shit',
    date: '2020-07-15',
    content: 'asdasdad',
    tags: [
      { id: 6, name: 'Work', color: '#9C27B0' },
      { id: 7, name: 'Coding', color: '#FF9800' },
      { id: 8, name: 'Stuff', color: '#4CAF50' }
    ]
  },
  {
    id: 5,
    title: 'Post 129',
    date: '2020-05-07',
    content: 'asdasdad'
  }
]
