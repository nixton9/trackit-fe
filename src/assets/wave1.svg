<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="744" height="744" viewBox="0 0 744 744">
  <defs>
    <clipPath id="clip-Prancheta_7">
      <rect width="744" height="744"/>
    </clipPath>
  </defs>
  <g id="Prancheta_7" data-name="Prancheta â€“ 7" clip-path="url(#clip-Prancheta_7)">
    <path id="Caminho_294" data-name="Caminho 294" d="M-14697.143-11591.146s204.548,93.02,241.44,226.007,101.77,246.085,225.016,271.63,239.81-133.94,286.815-109.888,72.048,136.969,72.048,136.969,18.206,112.27,0,146.063-154.687,113.568-154.687,113.568-145.483,41.1-252.11,41.649-282.684-4.542-332.2-22.729-57.569-16.607-86.319-46.7-74.028-81.864-81.211-123.045-11.661-111.631-8.345-135.894,30.468-240.38,30.94-257.871-.572-130.1,0-136.284S-14697.143-11591.146-14697.143-11591.146Z" transform="translate(14695 11649)" fill="#273150" opacity="0.051"/>
  </g>
</svg>
