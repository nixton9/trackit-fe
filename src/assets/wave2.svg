<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="744" height="744" viewBox="0 0 744 744">
  <defs>
    <clipPath id="clip-Prancheta_7">
      <rect width="744" height="744"/>
    </clipPath>
  </defs>
  <g id="Prancheta_7" data-name="Prancheta â€“ 7" clip-path="url(#clip-Prancheta_7)">
    <path id="Caminho_307" data-name="Caminho 307" d="M-14243.049-11762.149s99.251-140.3,221.355-63.8,71.256,212.9,188.94,256.211,284.029-180.642,371.026-162.149,131.877,179.9,131.877,179.9L-13407-11443.72l-79.384,66.653s-188.631,35.67-195.492,37.041-321.083,20.295-351.518,20.295-209.207-15.854-221.288-20.295-32.708-292.639-32.708-292.639Z" transform="translate(14215 12160.858)" fill="#273150" opacity="0.041"/>
  </g>
</svg>
