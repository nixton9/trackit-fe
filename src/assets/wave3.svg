<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="744" height="744" viewBox="0 0 744 744">
  <defs>
    <clipPath id="clip-Prancheta_7">
      <rect width="744" height="744"/>
    </clipPath>
  </defs>
  <g id="Prancheta_7" data-name="Prancheta â€“ 7" clip-path="url(#clip-Prancheta_7)">
    <path id="Caminho_308" data-name="Caminho 308" d="M1430.13,18293.221s356.522-204.141,537.172-65.336,273.16-352.586,456.213-279.107,399.449,358.18,399.449,358.18,205.3,357.1,208.9,365.451-160.994,153.961-208.9,147.217-546.637-79.807-705.6-97.068-577.556-19.437-656.578-28.982-318.938-34.217-335.6-53.873,280.937-332.744,280.937-332.744Z" transform="translate(-1614 -17637)" fill="#273150" opacity="0.04"/>
  </g>
</svg>
