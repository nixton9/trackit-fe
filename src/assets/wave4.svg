<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="744" height="744" viewBox="0 0 744 744">
  <defs>
    <clipPath id="clip-Prancheta_7">
      <rect width="744" height="744"/>
    </clipPath>
  </defs>
  <g id="Prancheta_7" data-name="Prancheta â€“ 7" clip-path="url(#clip-Prancheta_7)">
    <path id="Caminho_307" data-name="Caminho 307" d="M1449.88,18572s112.775-319.779,271.458-247.187,168.861,179.166,303.627,151.676,170.134-339.822,339.459-310.615,218.6,83.457,218.6,83.457l52.227,227.158v247.463s-162.445,46.264-172.641,44.359-378.379,27.41-395.417,29.01-439.332,0-439.332,0l-251.667-29.01Z" transform="translate(-1613 -17892)" fill="#273150" opacity="0.04"/>
  </g>
</svg>
