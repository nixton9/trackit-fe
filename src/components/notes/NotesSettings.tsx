import React, { useState, useEffect, Dispatch, SetStateAction } from 'react'
import Drawer from '../misc/Drawer'
import Tag from './Tag'
import { TagEditor } from './TagEditor'
import { SelectMenu } from '../misc/SelectMenu'
import { NotificationTypes, notificationState } from '../misc/Notification'
import { alertState } from '../misc/Alert'
import { Styled } from '../../styles/Settings.styles'
import { TAGS, NOTES } from '../../utils/queries'
import { useToggleElement } from '../../utils/useToggleElement'
import { ReactComponent as SettingsIcon } from '../../assets/icons/settings.svg'
import { ReactComponent as PlusIcon } from '../../assets/icons/plus.svg'
import { NoteTag } from '../../utils/ModuleTypes'
import { SortBySettings } from '../../utils/SettingsTypes'
import { useSetRecoilState } from 'recoil'
import { gql, useMutation, useQuery } from '@apollo/client'
import { useHistory } from 'react-router-dom'

const DELETE_TAG = gql`
  mutation DeleteTag($id: ID!) {
    deleteTag(id: $id) {
      id_tag
    }
  }
`

const sortByOptions = [
  { val: SortBySettings.DATE, label: 'Date' },
  { val: SortBySettings.ALPHABETICAL, label: 'Alphabetical Order' }
]

type NotesSettingsProps = {
  tags: NoteTag[]
  sortBy: string
  setSortBy: Dispatch<SetStateAction<SortBySettings>>
}

const NotesSettings: React.FC<NotesSettingsProps> = ({
  tags,
  sortBy,
  setSortBy
}) => {
  const [showTagEditor, setShowTagEditor] = useState(false)
  const [activeTag, setActiveTag] = useState<NoteTag | null>(null)

  const setNotification = useSetRecoilState(notificationState)
  const setAlert = useSetRecoilState(alertState)

  const history = useHistory()

  const [open, setOpen, overlayEl] = useToggleElement(() =>
    setShowTagEditor(false)
  )

  const { refetch: refetchTags } = useQuery(TAGS, {
    fetchPolicy: 'network-only'
  })
  const { refetch: refetchNotes } = useQuery(NOTES)

  const [deleteTag] = useMutation(DELETE_TAG)

  const handleSortByChange = (e: React.ChangeEvent<HTMLSelectElement>) =>
    setSortBy(e.target.value as SortBySettings)

  const handleTagClick = (tag: NoteTag) => {
    setActiveTag(tag)
    setShowTagEditor(true)
  }

  const handlePlusClick = () => {
    setActiveTag(null)
    setShowTagEditor(true)
  }

  const handleDeleteTag = (tagId: string | number) => {
    deleteTag({
      variables: {
        id: tagId
      }
    })
      .then(res => {
        setNotification({
          text: 'Tag was deleted',
          type: NotificationTypes.Success
        })
        refetchTags()
        refetchNotes()
      })
      .catch(err =>
        setNotification({
          text: 'There was a problem, please try again',
          type: NotificationTypes.Error
        })
      )
  }

  const handleDeleteTagConfirm = (tagId: string | number) => {
    setShowTagEditor(false)
    setAlert({
      text: 'This tag will be removed.',
      onConfirm: () => handleDeleteTag(tagId)
    })
  }

  useEffect(() => {
    history.push(`#`)

    window.onpopstate = (e: any) => {
      setOpen(false)
    }
  }, [history, setOpen])

  return (
    <>
      <SettingsIcon
        className="settings-icon"
        onClick={() => setOpen(true)}
        data-test-id="notes-settings-icon"
      />

      <Drawer title="Settings" open={open} overlayRef={overlayEl}>
        <>
          <Styled.SettingsBlock>
            <Styled.SettingsBlock__Label htmlFor="sort-by">
              Sort by
            </Styled.SettingsBlock__Label>
            <Styled.SettingsBlock__Input>
              <SelectMenu
                id="sort-by"
                value={sortBy}
                onChange={handleSortByChange}
                options={sortByOptions}
              />
            </Styled.SettingsBlock__Input>
          </Styled.SettingsBlock>

          <Styled.SettingsBlock>
            <Styled.SettingsBlock__Label>Tags</Styled.SettingsBlock__Label>
            <Styled.SettingsBlock__Tags>
              {tags &&
                tags.map(tag => (
                  <Tag
                    key={tag.id}
                    id={tag.id}
                    name={tag.name}
                    color={tag.color}
                    onClick={() => handleTagClick(tag)}
                    onDelete={() => handleDeleteTagConfirm(tag.id)}
                  />
                ))}
              <Styled.SettingsBlock__Icon
                onClick={handlePlusClick}
                className="mbl-click"
                data-test-id="tags-add-icon"
              >
                <PlusIcon />
              </Styled.SettingsBlock__Icon>
            </Styled.SettingsBlock__Tags>
          </Styled.SettingsBlock>

          {showTagEditor && (
            <TagEditor
              tag={activeTag}
              closeEditor={() => setShowTagEditor(false)}
            />
          )}
        </>
      </Drawer>
    </>
  )
}

export default NotesSettings
