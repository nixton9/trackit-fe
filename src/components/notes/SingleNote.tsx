import React from 'react'
import { TagChip } from './TagChip'
import { Styled } from '../../styles/Single.styles'
import { Note } from '../../utils/ModuleTypes'
import { ReactComponent as CalendarIcon } from '../../assets/icons/calendar.svg'
import { NavLink } from 'react-router-dom'
import { displayDateString, parseDateInverse } from '../../utils/dateHelpers'

const SingleNote: React.FC<Note> = ({ id, title, date, tags }) => {
  return (
    <Styled.SingleContainer className="single-note">
      <NavLink to={`/notes/${id}`}>
        <Styled.SingleFlex>
          <div className="note-info">
            <Styled.SingleTitle className="note-title">
              {title}
            </Styled.SingleTitle>
            <Styled.SingleDate>
              <CalendarIcon />
              <p>{displayDateString(parseDateInverse(date))}</p>
            </Styled.SingleDate>
          </div>
          <Styled.SingleNote__Tags>
            {tags &&
              tags.map(tag => (
                <TagChip
                  key={tag.id}
                  id={tag.id}
                  name={tag.name}
                  color={tag.color}
                />
              ))}
          </Styled.SingleNote__Tags>
        </Styled.SingleFlex>
      </NavLink>
    </Styled.SingleContainer>
  )
}

export default SingleNote
