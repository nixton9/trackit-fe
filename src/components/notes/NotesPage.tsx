import React, { useState } from 'react'
import NotesSettings from './NotesSettings'
import SingleNote from './SingleNote'
import { SelectMenu } from '../misc/SelectMenu'
import { PageLoading } from '../misc/PageLoading'
import { PageError } from '../misc/PageError'
import { activeContentState } from '../misc/Add'
import { Styled } from '../../styles/Page.styles'
import { Note, NoteTag, ModuleTypes } from '../../utils/ModuleTypes'
import { NOTES, TAGS } from '../../utils/queries'
import { SortBySettings } from '../../utils/SettingsTypes'
import { sortData } from '../../utils/globalHelpers'
import { ReactComponent as PlusIcon } from '../../assets/icons/plus.svg'
import { ReactComponent as NoDataIcon } from '../../assets/icons/nodata.svg'
import { useQuery } from '@apollo/client'
import { useSetRecoilState } from 'recoil'

const NotesPage: React.FC = () => {
  const setActiveContent = useSetRecoilState(activeContentState)

  const { loading, error, data } = useQuery(NOTES)
  const { data: tags } = useQuery(TAGS, {
    fetchPolicy: 'no-cache'
  })

  const [view, setView] = useState('all')
  const [sortBy, setSortBy] = useState<SortBySettings>(SortBySettings.DATE)

  const viewOptions = tags
    ? [
        { val: 'all', label: 'All' },
        ...tags.tags.map((tag: NoteTag) => ({
          val: tag.id,
          label: tag.name
        }))
      ]
    : [{ val: 'all', label: 'All' }]

  const handleViewChange = (e: React.ChangeEvent<HTMLSelectElement>) =>
    setView(e.target.value)

  const visibleNotes = data
    ? view === 'all'
      ? data.notes
      : data.notes.filter(
          (note: Note) =>
            note.tags?.filter(tag => Number(tag.id) === Number(view)).length
        )
    : []

  const sortedNotes = sortData(visibleNotes, sortBy)

  return (
    <Styled.PageContainer>
      <Styled.PageTitle>Notes</Styled.PageTitle>

      <Styled.PageHeader>
        <Styled.PageHeader__View>
          <Styled.PageHeader__View__Dropdown>
            <SelectMenu
              id="notes-view"
              value={view}
              onChange={handleViewChange}
              options={viewOptions}
              itemClass={'view-select-item'}
            />
          </Styled.PageHeader__View__Dropdown>
          <Styled.PageHeader__View__Counter>
            {data ? data.notes.length : ''}
          </Styled.PageHeader__View__Counter>
        </Styled.PageHeader__View>
        <Styled.PageHeader__Settings>
          <NotesSettings
            tags={tags ? tags.tags : []}
            sortBy={sortBy}
            setSortBy={setSortBy}
          />
        </Styled.PageHeader__Settings>
      </Styled.PageHeader>

      <Styled.PageContent>
        {error ? (
          <PageError>{error.message}</PageError>
        ) : loading ? (
          <PageLoading />
        ) : sortedNotes.length ? (
          (sortedNotes as Note[]).map(note => (
            <SingleNote
              key={note.id}
              id={note.id}
              title={note.title}
              date={note.date}
              tags={note.tags}
            />
          ))
        ) : (
          <Styled.PageContent__NoData>
            <NoDataIcon />
          </Styled.PageContent__NoData>
        )}
      </Styled.PageContent>
      <Styled.PageAddItem onClick={() => setActiveContent(ModuleTypes.Notes)}>
        <PlusIcon />
      </Styled.PageAddItem>
    </Styled.PageContainer>
  )
}

export default NotesPage
