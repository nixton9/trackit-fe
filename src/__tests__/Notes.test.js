import React from 'react'
import { render } from '@testing-library/react'
import NotesPage from '../components/notes/NotesPage'
import SingleNote from '../components/notes/SingleNote'
import NotesSettings from '../components/notes/NotesSettings'
import { notes, notesTags } from '../assets/fakeData'
import { BrowserRouter } from 'react-router-dom'
import { ApolloClient, ApolloProvider, InMemoryCache } from '@apollo/client'

const client = new ApolloClient({
  uri: '/',
  cache: new InMemoryCache(),
  headers: {
    authorization: `Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOjYsImlhdCI6MTU5NDYyMzYxN30.eRuuyy-6EE6UkGFQI9UQmRklwPiyXgSJgmoZektpiG4`
  }
})

describe('Notes Page', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <ApolloProvider client={client}>
        <BrowserRouter>
          <NotesPage />
        </BrowserRouter>
      </ApolloProvider>
    )

    expect(container).toMatchSnapshot()
  })

  it('has the title', () => {
    const { getByText } = render(
      <ApolloProvider client={client}>
        <BrowserRouter>
          <NotesPage />
        </BrowserRouter>
      </ApolloProvider>
    )

    expect(getByText('Notes')).toBeInTheDocument()
  })
})

describe('SingleNote', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <SingleNote
          id={notes[0].id}
          title={notes[0].title}
          date={notes[0].date}
          tags={notes[0].tags}
        />
      </BrowserRouter>
    )

    expect(container).toMatchSnapshot()
  })

  it('has all the content of the note', () => {
    const { getByText } = render(
      <BrowserRouter>
        <SingleNote
          id={notes[0].id}
          title={notes[0].title}
          date={notes[0].date}
          tags={notes[0].tags}
        />
      </BrowserRouter>
    )

    expect(getByText('Lista de compras')).toBeInTheDocument()
    expect(getByText('20 Jul')).toBeInTheDocument()
    expect(getByText('Primary')).toBeInTheDocument()
    expect(getByText('Secondary')).toBeInTheDocument()
  })
})

describe('NotesSettings', () => {
  it('matches snashot', () => {
    const { container } = render(<NotesSettings tags={notesTags} />)

    expect(container).toMatchSnapshot()
  })

  it('has all the fields', () => {
    const { getByText } = render(<NotesSettings tags={notesTags} />)

    expect(getByText('Sort by')).toBeInTheDocument()
    expect(getByText('Font size')).toBeInTheDocument()
    expect(getByText('Tags')).toBeInTheDocument()
  })

  it('shows tags', () => {
    const { getByText } = render(<NotesSettings tags={notesTags} />)

    expect(getByText('Primary')).toBeInTheDocument()
    expect(getByText('Secondary')).toBeInTheDocument()
    expect(getByText('Work')).toBeInTheDocument()
    expect(getByText('Stuff')).toBeInTheDocument()
  })
})
