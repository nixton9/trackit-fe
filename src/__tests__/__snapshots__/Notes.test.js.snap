// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notes Page matches snapshot 1`] = `
<div>
  <div
    class="sc-fzoXzr jphFpM"
  >
    <h1
      class="sc-fzpjYC lnMTjC"
    >
      Notes
    </h1>
    <div
      class="sc-fznxsB cvPTFV"
    >
      <div
        class="sc-fznJRM fUFvhP"
      >
        <div
          class="sc-fznWqX cfsIUA"
        >
          <div
            class="MuiInputBase-root MuiInput-root"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="notes-view"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="notes-view"
              role="button"
              tabindex="0"
            >
              All
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="all"
            />
            <svg
              class="MuiSelect-icon"
            >
              chevron.svg
            </svg>
          </div>
        </div>
        <h3
          class="sc-fzoiQi FHcDk"
        >
          5
        </h3>
      </div>
      <div
        class="sc-fzqARJ irCVwi"
      >
        <svg
          class="settings-icon"
        >
          settings.svg
        </svg>
        <div
          class="sc-AxjAm khxSQh"
        >
          <h2
            class="sc-AxiKw jzoYkp"
          >
            Settings
          </h2>
          <div
            class="sc-AxhCb iGvJwE"
          >
            <div
              class="sc-AxgMl ibGeUx"
            >
              <label
                class="sc-AxheI btNIfl"
                for="sort-by"
              >
                Sort by
              </label>
              <div
                class="sc-Axmtr dWIuPP"
              >
                <div
                  class="MuiInputBase-root MuiInput-root"
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="sort-by"
                    class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                    id="sort-by"
                    role="button"
                    tabindex="0"
                  >
                    Date
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput"
                    tabindex="-1"
                    value="date"
                  />
                  <svg
                    class="MuiSelect-icon"
                  >
                    chevron.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="sc-AxgMl ibGeUx"
            >
              <label
                class="sc-AxheI btNIfl"
                for="font-size"
              >
                Font size
              </label>
              <div
                class="sc-Axmtr dWIuPP"
              >
                <div
                  class="MuiInputBase-root MuiInput-root"
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="font-size"
                    class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                    id="font-size"
                    role="button"
                    tabindex="0"
                  >
                    Medium
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput"
                    tabindex="-1"
                    value="medium"
                  />
                  <svg
                    class="MuiSelect-icon"
                  >
                    chevron.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="sc-AxgMl ibGeUx"
            >
              <label
                class="sc-AxheI btNIfl"
              >
                Tags
              </label>
              <div
                class="sc-AxmLO fFgNoy"
              >
                <div
                  class="sc-AxhUy kRHjcT single-tag"
                  color="#F44336"
                >
                  <span>
                    Primary
                  </span>
                </div>
                <div
                  class="sc-AxhUy gESsUG single-tag"
                  color="#E91E63"
                >
                  <span>
                    Secondary
                  </span>
                </div>
                <div
                  class="sc-AxhUy hEWzsc single-tag"
                  color="#607D8B"
                >
                  <span>
                    Studies
                  </span>
                </div>
                <div
                  class="sc-AxhUy kIetSe single-tag"
                  color="#00BCD4"
                >
                  <span>
                    School
                  </span>
                </div>
                <div
                  class="sc-AxhUy xeezF single-tag"
                  color="#3F51B5"
                >
                  <span>
                    Random
                  </span>
                </div>
                <div
                  class="sc-AxhUy ktyFwq single-tag"
                  color="#9C27B0"
                >
                  <span>
                    Work
                  </span>
                </div>
                <div
                  class="sc-AxhUy gdgqSe single-tag"
                  color="#FF9800"
                >
                  <span>
                    Coding
                  </span>
                </div>
                <div
                  class="sc-AxhUy glbvAi single-tag"
                  color="#4CAF50"
                >
                  <span>
                    Stuff
                  </span>
                </div>
                <span
                  class="sc-fzpans eLxAbI"
                >
                  <svg>
                    plus.svg
                  </svg>
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-AxirZ gpexXZ"
        />
      </div>
    </div>
    <section
      class="sc-fzqNqU cgxesj"
    >
      <article
        class="sc-fzplWN fBwfkr"
      >
        <a
          href="/notes/1"
        >
          <div
            class="sc-fznyAO lkMwHr"
          >
            <div>
              <h5
                class="sc-fznKkj cEaWPO"
              >
                Lista de compras
              </h5>
              <p
                class="sc-fznZeY kCMyLg"
              >
                20 Jul
              </p>
            </div>
            <div
              class="sc-fzqBZW fqzjjU"
            >
              <div
                class="sc-AxhUy kRHjcT single-tag"
                color="#F44336"
              >
                <span>
                  Primary
                </span>
              </div>
              <div
                class="sc-AxhUy gESsUG single-tag"
                color="#E91E63"
              >
                <span>
                  Secondary
                </span>
              </div>
            </div>
          </div>
        </a>
      </article>
      <article
        class="sc-fzplWN fBwfkr"
      >
        <a
          href="/notes/2"
        >
          <div
            class="sc-fznyAO lkMwHr"
          >
            <div>
              <h5
                class="sc-fznKkj cEaWPO"
              >
                Apontamentos aula
              </h5>
              <p
                class="sc-fznZeY kCMyLg"
              >
                25 Jun
              </p>
            </div>
            <div
              class="sc-fzqBZW fqzjjU"
            >
              <div
                class="sc-AxhUy kIetSe single-tag"
                color="#00BCD4"
              >
                <span>
                  School
                </span>
              </div>
            </div>
          </div>
        </a>
      </article>
      <article
        class="sc-fzplWN fBwfkr"
      >
        <a
          href="/notes/3"
        >
          <div
            class="sc-fznyAO lkMwHr"
          >
            <div>
              <h5
                class="sc-fznKkj cEaWPO"
              >
                Random shit
              </h5>
              <p
                class="sc-fznZeY kCMyLg"
              >
                15 Jul
              </p>
            </div>
            <div
              class="sc-fzqBZW fqzjjU"
            >
              <div
                class="sc-AxhUy xeezF single-tag"
                color="#3F51B5"
              >
                <span>
                  Random
                </span>
              </div>
            </div>
          </div>
        </a>
      </article>
      <article
        class="sc-fzplWN fBwfkr"
      >
        <a
          href="/notes/4"
        >
          <div
            class="sc-fznyAO lkMwHr"
          >
            <div>
              <h5
                class="sc-fznKkj cEaWPO"
              >
                Work shit
              </h5>
              <p
                class="sc-fznZeY kCMyLg"
              >
                15 Jul
              </p>
            </div>
            <div
              class="sc-fzqBZW fqzjjU"
            >
              <div
                class="sc-AxhUy ktyFwq single-tag"
                color="#9C27B0"
              >
                <span>
                  Work
                </span>
              </div>
              <div
                class="sc-AxhUy gdgqSe single-tag"
                color="#FF9800"
              >
                <span>
                  Coding
                </span>
              </div>
              <div
                class="sc-AxhUy glbvAi single-tag"
                color="#4CAF50"
              >
                <span>
                  Stuff
                </span>
              </div>
            </div>
          </div>
        </a>
      </article>
      <article
        class="sc-fzplWN fBwfkr"
      >
        <a
          href="/notes/5"
        >
          <div
            class="sc-fznyAO lkMwHr"
          >
            <div>
              <h5
                class="sc-fznKkj cEaWPO"
              >
                Post 129
              </h5>
              <p
                class="sc-fznZeY kCMyLg"
              >
                03 Aug
              </p>
            </div>
            <div
              class="sc-fzqBZW fqzjjU"
            />
          </div>
        </a>
      </article>
    </section>
  </div>
</div>
`;

exports[`NotesSettings matches snashot 1`] = `
<div>
  <svg
    class="settings-icon"
  >
    settings.svg
  </svg>
  <div
    class="sc-AxjAm khxSQh"
  >
    <h2
      class="sc-AxiKw jzoYkp"
    >
      Settings
    </h2>
    <div
      class="sc-AxhCb iGvJwE"
    >
      <div
        class="sc-AxgMl ibGeUx"
      >
        <label
          class="sc-AxheI btNIfl"
          for="sort-by"
        >
          Sort by
        </label>
        <div
          class="sc-Axmtr dWIuPP"
        >
          <div
            class="MuiInputBase-root MuiInput-root"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="sort-by"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="sort-by"
              role="button"
              tabindex="0"
            >
              Date
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="date"
            />
            <svg
              class="MuiSelect-icon"
            >
              chevron.svg
            </svg>
          </div>
        </div>
      </div>
      <div
        class="sc-AxgMl ibGeUx"
      >
        <label
          class="sc-AxheI btNIfl"
          for="font-size"
        >
          Font size
        </label>
        <div
          class="sc-Axmtr dWIuPP"
        >
          <div
            class="MuiInputBase-root MuiInput-root"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="font-size"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="font-size"
              role="button"
              tabindex="0"
            >
              Medium
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="medium"
            />
            <svg
              class="MuiSelect-icon"
            >
              chevron.svg
            </svg>
          </div>
        </div>
      </div>
      <div
        class="sc-AxgMl ibGeUx"
      >
        <label
          class="sc-AxheI btNIfl"
        >
          Tags
        </label>
        <div
          class="sc-AxmLO fFgNoy"
        >
          <div
            class="sc-AxhUy kRHjcT single-tag"
            color="#F44336"
          >
            <span>
              Primary
            </span>
          </div>
          <div
            class="sc-AxhUy gESsUG single-tag"
            color="#E91E63"
          >
            <span>
              Secondary
            </span>
          </div>
          <div
            class="sc-AxhUy hEWzsc single-tag"
            color="#607D8B"
          >
            <span>
              Studies
            </span>
          </div>
          <div
            class="sc-AxhUy kIetSe single-tag"
            color="#00BCD4"
          >
            <span>
              School
            </span>
          </div>
          <div
            class="sc-AxhUy xeezF single-tag"
            color="#3F51B5"
          >
            <span>
              Random
            </span>
          </div>
          <div
            class="sc-AxhUy ktyFwq single-tag"
            color="#9C27B0"
          >
            <span>
              Work
            </span>
          </div>
          <div
            class="sc-AxhUy gdgqSe single-tag"
            color="#FF9800"
          >
            <span>
              Coding
            </span>
          </div>
          <div
            class="sc-AxhUy glbvAi single-tag"
            color="#4CAF50"
          >
            <span>
              Stuff
            </span>
          </div>
          <span
            class="sc-fzpans eLxAbI"
          >
            <svg>
              plus.svg
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="sc-AxirZ gpexXZ"
  />
</div>
`;

exports[`SingleNote matches snapshot 1`] = `
<div>
  <article
    class="sc-fzplWN fBwfkr"
  >
    <a
      href="/notes/1"
    >
      <div
        class="sc-fznyAO lkMwHr"
      >
        <div>
          <h5
            class="sc-fznKkj cEaWPO"
          >
            Lista de compras
          </h5>
          <p
            class="sc-fznZeY kCMyLg"
          >
            20 Jul
          </p>
        </div>
        <div
          class="sc-fzqBZW fqzjjU"
        >
          <div
            class="sc-AxhUy kRHjcT single-tag"
            color="#F44336"
          >
            <span>
              Primary
            </span>
          </div>
          <div
            class="sc-AxhUy gESsUG single-tag"
            color="#E91E63"
          >
            <span>
              Secondary
            </span>
          </div>
        </div>
      </div>
    </a>
  </article>
</div>
`;
